### DISCUSSION TITLE
Break out of display manager tweaking, facility X2Go TCE MiniDesktop

### DISCUSSION THREAD

**Discussion by sunweaver** [ltsp/ltsp#213] [Q&A]
Hi Alkis, hi all,

today I deployed LTSP 19.11 (as currently in Debian unstable) as a Proof-of-Concept at a customer of mine and combined it with a package from X2Go: x2gothinclient-minidesktop [1].

The X2Go TCE minidesktop package already configures LightDM, so that it boots into a pre-configured account and runs a certain set of session initialization scripts [2,3].

As LTSP tweaks lightdm.conf itself, I had to add a patch to LTSP init/client/55-display-manager.sh:

```
[mike@minobo ltsp (master)]$ cat no-display-manager-tweaks.patch 
--- 55-display-manager.sh.orig  2019-11-26 16:02:42.248981444 +0100
+++ 55-display-manager.sh       2019-11-26 14:45:50.775820603 +0100
@@ -11,6 +11,9 @@
 display_manager_main() {
     local _ALUSER _AUTOLOGIN _RELOGIN host
 
+    if [ -n "$NO_DISPLAY_MANAGER_TWEAKS" ]; then
+        return 0
+    fi
     if [ -z "$AUTOLOGIN" ]; then
         _AUTOLOGIN=false
         _ALUSER=

```

Would such a new ltsp.conf parameter be a valid solution to you? Or should I drop a script (e.g. 90-display-manager.sh and override the ``display_manager_main()`` function and bluntly make it a no-op function?

I will see to getting the X2Go TCE to Debian during the bullseye release cycle, so people can more easily combine LTSP and X2Go TCE natively in Debian.

Requesting your feedback! Thanks,
Mike

PS: awesome good work that you did on the LTSP rewrite!!!

[1] https://code.x2go.org/gitweb?p=x2gothinclient.git;a=tree;f=minidesktop;h=91778346009c6a54da06b59bf6e29c1c016a0a60;hb=HEAD
[2] https://code.x2go.org/gitweb?p=x2gothinclient.git;a=blob;f=minidesktop/etc/lightdm.conf;h=b3f6f552e9b7006ba362cb151db08cdab4cc4f69;hb=HEAD
[3] https://code.x2go.org/gitweb?p=x2gothinclient.git;a=tree;f=management/share/etc/x2gothinclient_init.d;h=d26e86a64e415e5da2c31efa572787fe219d94e8;hb=HEAD

**Comment by alkisg** [comment]
Hi Gabriel, thank you for your good words and for your great work in all the packages you maintain,

does this work for you? It might not be there yet in the sid version:

```shell
OMIT_FUNCTIONS="display_manager_main"
```

See https://ltsp.org/man/ltsp.conf/ for its documentation.

**Comment by sunweaver** [comment]
Hi Alkis,

On  Di 26 Nov 2019 17:58:21 CET, Alkis Georgopoulos wrote:

> Hi Gabriel, thank you for your good words and for your great work in  
> all the packages you maintain,
>
> does this work for you? It might not be there yet in the sid version:
>
> ```shell
> OMIT_FUNCTIONS="display_manager_main"
> ```
>
> See https://ltsp.org/man/ltsp.conf/ for its documentation.

Ok, that might be an option, indeed. Thanks.

I will propose it to the customer.

For X2Go TCE MiniDesktop via LightDM. Is it possible to drop files  
into /etc/ltsp/ltsp.conf.d/ and have them processed in alpha-numerical  
order?

Greets,
Mike
-- 

DAS-NETZWERKTEAM
c\o Technik- und Ökologiezentrum Eckernförde
Mike Gabriel, Marienthaler str. 17, 24340 Eckernförde
mobile: +49 (1520) 1976 148
landline: +49 (4351) 850 8940

GnuPG Fingerprint: 9BFB AEE8 6C0A A5FF BF22  0782 9AF4 6B30 2577 1B31
mail: mike.gabriel@das-netzwerkteam.de, http://das-netzwerkteam.de

**Comment by alkisg** [comment]
(moved to the community issues that we use for discussions)

Check: https://github.com/ltsp/ltsp/blob/master/ltsp/ltsp#L214
If you drop files e.g. in /etc/ltsp/client/init/*, they'll be sourced in order, and may override parameters defaults or provide additional functions.
If it's a package though, it'd better drop them directly in /usr/share/ltsp/client...

**Comment by sunweaver** [comment]
Hi Alkis,

On  Di 26 Nov 2019 18:28:09 CET, Alkis Georgopoulos wrote:

> (moved to the community issues that we use for discussions)
>
> Check: https://github.com/ltsp/ltsp/blob/master/ltsp/ltsp#L214
> If you drop files e.g. in /etc/ltsp/client/init/*, they'll be  
> sourced in order, and may override or provide additional functions.
> If it's a package though, it'd better drop them directly in  
> /usr/share/ltsp/client...

Can you take a look at this commit?

https://code.x2go.org/gitweb?p=x2gothinclient.git;a=commitdiff;h=da350c91c5c2e2d9f60c6fb853386de052485c85

That should display the display_manager_main() function, right?

Mike
-- 

DAS-NETZWERKTEAM
c\o Technik- und Ökologiezentrum Eckernförde
Mike Gabriel, Marienthaler str. 17, 24340 Eckernförde
mobile: +49 (1520) 1976 148
landline: +49 (4351) 850 8940

GnuPG Fingerprint: 9BFB AEE8 6C0A A5FF BF22  0782 9AF4 6B30 2577 1B31
mail: mike.gabriel@das-netzwerkteam.de, http://das-netzwerkteam.de

**Comment by alkisg** [comment]
Hi Mike,

this should work fine; I suggest to number it 88- instead of 60- though, as I'm trying to reserve 1x and 8x for other packages, 2x, 3x, 4x, 5x, 6x, 7x for upstream LTSP, and 0x, 9x for local administrators. I have a bit of documentation in https://ltsp.org/advanced/coding/ about this, although it was written before the new LTSP was completed, so some parts aren't accurate atm.

**Comment by sunweaver** [comment]
Hi Alkis,

On  Mi 27 Nov 2019 09:16:42 CET, Alkis Georgopoulos wrote:

> Hi Mike,
>
> this should work fine; I suggest to number it 88- instead of 60-  
> though, as I'm trying to reserve 1x and 8x for other packages, 2x,  
> 3x, 4x, 5x, 6x, 7x for upstream LTSP, and 0x, 9x for local  
> administrators. I have a bit of documentation in  
> https://ltsp.org/advanced/coding/ about this, although it was  
> written before the new LTSP was completed, so some parts aren't  
> accurate atm.

ok. Done.

https://code.x2go.org/gitweb?p=x2gothinclient.git;a=commitdiff;h=e20ff930d5cfe62fde84d2aaf23d8c1d70872689

Mike

PS: please also take note of Debian bug #945521 [1]. That's a flaw  
that is at the moment only specific to Debian testing/unstable but  
will hit other distros (e.g. Ubuntu 20.04 LTS), too.

[1] https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=945521
-- 

DAS-NETZWERKTEAM
c\o Technik- und Ökologiezentrum Eckernförde
Mike Gabriel, Marienthaler str. 17, 24340 Eckernförde
mobile: +49 (1520) 1976 148
landline: +49 (4351) 850 8940

GnuPG Fingerprint: 9BFB AEE8 6C0A A5FF BF22  0782 9AF4 6B30 2577 1B31
mail: mike.gabriel@das-netzwerkteam.de, http://das-netzwerkteam.de

**Comment by alkisg** [comment]
Mike please file LTSP bugs upstream, I'm trying to keep it as distro-independent as possible.

Distribution bugs should be filed only for backports of upstream fixes.

Regarding nonempty, thanks for the heads up, it looks like [it's now on by default](https://github.com/libfuse/libfuse/commit/0bef21e8), so we need a version check in order to be consistent with previous releases.

It would be best if a bug report was filed against fuse itself to allow `-o nonempty` and ignore it for a few years, in order to not break existing software.
